### PR TITLE
fabtests/pytest: Skip cuda tests if provider does not support hmem hints

### PR DESCRIPTION
**Pull request by wckzhang**
Signed-off-by: William Zhang <wilzhang@amazon.com>

### CODE REVIEW

File: fabtests/Makefile.am [hunk 1]
```diff
@@ -454,6 +455,14 @@ regression_sighandler_test_LDADD = libfabtests.la
 regression_sighandler_test_CFLAGS = \
 	$(AM_CFLAGS)
 
+common_check_hmem_SOURCES = \
+	common/check_hmem.c
```
Comment (shijin-aws) [inline fabtests/Makefile.am:459]: can we directly use the `fi_info` bin from libfabric?

Comment (wckzhang) [reply fabtests/Makefile.am:459]: we could, but I wasn't sure if there was a way to guarantee it was the libfabric that fabtests was built against. There's probably a way to do so I can try looking into it

Comment (wckzhang) [reply fabtests/Makefile.am:459]: I'd guess we can use the configure --prefix info, I'm not sure though if we can pass the configure/prefix info to python

File: fabtests/common/check_hmem.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * This test returns whether or not efa supports FI_HMEM
```
Comment (tschuett) [inline fabtests/common/check_hmem.c:29]: This works for any provider. The name is passed as `argv[1]`.

File: fabtests/pytest/common.py [hunk 3]
```diff
@@ -1,9 +1,15 @@
 import pytest
 import errno
 
-def has_cuda(ip):
+def has_cuda(cmdline_args, host_type, ip):
```
Comment (wzamazon) [inline fabtests/pytest/common.py:4]: the function name should be changed to `has_hmem_support`.

File: fabtests/pytest/common.py [hunk 4]
```diff
@@ -198,14 +204,14 @@ def prepare_base_command(self, command_type, executable,
         # in communication test, client is sender, server is receiver
         client_memory_type,server_memory_type = memory_type.split("_to_")
         if command_type == "server" and server_memory_type == "cuda":
-            if not has_cuda(self._cmdline_args.server_id):
+            if not has_cuda(self._cmdline_args, "server", self._cmdline_args.server_id):
                 pytest.skip("no cuda device")
```
Comment (wzamazon) [inline fabtests/pytest/common.py:208]: The skip message should be updated to `no hmem support`.

File: fabtests/Makefile.am (hunk 1 above)
Comment (shijin-aws) [reply fabtests/Makefile.am:459]: Yeah, I think it is not straightforward to use it. Do not bother then.

File: fabtests/pytest/common.py [hunk 5]
```diff
@@ -1,9 +1,15 @@
 import pytest
 import errno
 
-def has_cuda(ip):
+def has_cuda_hmem_support(cmdline_args, host_type, ip):
```
Comment (shijin-aws) [inline fabtests/pytest/common.py:4]: Can we split this function into two, `has_cuda` and `has_hmem_support`? In the future we will implement similar things for other hmem ifaces and they can reuse `has_hmem_support` .

Comment (wckzhang) [reply fabtests/pytest/common.py:4]: Sure

**Review by shijin-aws — APPROVED** [review]

**Comment by wckzhang** [comment]
Failure looks unrelated

**Comment by wckzhang** [comment]
Can someone help me merge this patch?

**Pull request merged**